browser/overlay-ports/protobuf/fix-arm64-msvc.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/src/google/protobuf/parse_context.h b/src/google/protobuf/parse_context.h
+index df12ee1ab..3eb2e56c7 100644
+--- a/src/google/protobuf/parse_context.h
++++ b/src/google/protobuf/parse_context.h
+@@ -653,7 +653,7 @@ inline const char* VarintParseSlow(const char* p, uint32_t res, uint64_t* out) {
+   return tmp.first;
+ }
+ 
+-#ifdef __aarch64__
++#if defined(__aarch64__) && !defined(_MSC_VER)
+ // Generally, speaking, the ARM-optimized Varint decode algorithm is to extract
+ // and concatenate all potentially valid data bits, compute the actual length
+ // of the Varint, and mask off the data bits which are not actually part of the
+@@ -883,7 +883,7 @@ static const char* VarintParseSlowArm(const char* p, uint64_t* out,
+ 
+ template <typename T>
+ PROTOBUF_NODISCARD const char* VarintParse(const char* p, T* out) {
+-#if defined(__aarch64__) && defined(PROTOBUF_LITTLE_ENDIAN)
++#if defined(__aarch64__) && defined(PROTOBUF_LITTLE_ENDIAN) && !defined(_MSC_VER)
+   // This optimization is not supported in big endian mode
+   uint64_t first8;
+   std::memcpy(&first8, p, sizeof(first8));
```

Comment:
what does MSVC not handle here? don't really mind, just curious/confused

Reply:
Not sure! This patch was not made by me, it's one of the included with vcpkg. Mine is just the pthreads one.
Here is the related commit: https://github.com/microsoft/vcpkg/pull/40173

Can probably be removed from our repo